Test: Add e2e test for 404 page

Infinite-Null opened the pull request:
Related to: https://github.com/WordPress/gutenberg/pull/69234#issuecomment-2671616050
PR: #69234

## What?
Add E2E test for 404 page in the Site Editor

## Why?
This PR adds coverage for 404 page in the Site Editor.

## Testing Instructions
1. Run E2E tests for the site-editor:
```
npm run test:e2e:debug -- test/e2e/specs/site-editor/navigation.spec.js
```
2. Verify the new 404 test passes

## Screenshots:

![image](https://github.com/user-attachments/assets/93f54081-c1ac-4daf-b8ae-ec4598e86998)

Infinite-Null committed:
Add e2e test for 404 page

Infinite-Null commented:
Hi @t-hamano,
I've added the E2E test for the 404 page in the Site Editor. Please review it at your convenience.
Thank You!

t-hamano commented on test/e2e/specs/site-editor/navigation.spec.js:
```diff
@@ -106,6 +106,25 @@ test.describe( 'Site editor navigation', () => {
+		await admin.visitSiteEditor();
+
+		// Navigate to a non-existent template.
+		await page.goto( '/wp-admin/site-editor.php?p=%2Ftemplate-foo-bar' );
```
```suggestion
		await admin.visitAdminPage( 'site-editor.php', 'p=/template-foo-bar' );
```

Let's use the utility.

t-hamano commented on test/e2e/specs/site-editor/navigation.spec.js:
```diff
@@ -106,6 +106,25 @@ test.describe( 'Site editor navigation', () => {
+		await page.goto( '/wp-admin/site-editor.php?p=%2Ftemplate-foo-bar' );
+
+		// Verify the 404 error notice is displayed with the correct message.
+		await expect(
+			page
+				.locator( '.components-notice__content' )
+				.getByText(
+					'The requested page could not be found. Please check the URL.'
+				)
+		).toBeVisible();
```
```suggestion
		await expect(
			page.locator(
				'.edit-site-layout__area .components-notice__content'
			)
		).toHaveText(
			'The requested page could not be found. Please check the URL.'
		);
```

We should be able to combine `getByText` and `toBeVisible` using `toHaveText`.

Infinite-Null committed:
Improve navigation test

Infinite-Null commented:
Hi @t-hamano,
Thank you for the feedback! I have implemented the changes. Please review it at your convenience.

t-hamano commented on test/e2e/specs/site-editor/navigation.spec.js:
```diff
@@ -106,6 +106,25 @@ test.describe( 'Site editor navigation', () => {
+		admin,
+		page,
+	} ) => {
+		await admin.visitSiteEditor();
+
+		// Navigate to a non-existent template.
```
```suggestion
		// Navigate to a non-existent template.
```

Nit: `visitSiteEditor()` is equivalent to `visitAdminPage( 'site-editor.php' )`, which I think can be removed since there is already a subsequent `visitAdminPage`.

Mamaduka replied:
I think that makes sense. What was the reason for calling `visitSiteEditor`?

Infinite-Null committed:
Remove unnecessery `visitSiteEditor()`

Infinite-Null commented:
Hi @t-hamano and @Mamaduka,
I have updated the PR and removed the `await admin.visitSiteEditor();`.
Please review it at your convenience.
Thank You!
